src/controllers/DrawingModeController.js

```diff
@@ -93,6 +93,7 @@ class DrawingModeController extends AnnotationModeController {
      */
     cancelDrawing(): void {
         if (this.currentThread) {
+            this.unregisterThread(this.currentThread);
```

Comment:
It looks like similar logic exists elsewhere in this file. Consider creating a `destroyThread` method to avoid this bug in the future.